City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-3451: High school search

How to test:

1. On you Kasvatus ja koulutus instance `make fresh`
2. Checkout this branch and checkout the branch with same name on your hdbt folder or run `composer require drupal/hdbt:dev-UHF-3451_high_school_search`
3. `make drush-cim && make drush-cr`
4. `make shell`
5. `drush locale:check && drush locale:update`
6. First go make sure you have two new vocabularies on the site: "Emphasis" and "Educational mission".
7. Add some terms to these vocabularies and add all three language translations for them.
8. Go to integrations page and make sure you have some units imported from TPR. Check if you have some high schools there so testing will be closer to real use case. Edit for example three to five high school units and add them Emphasis and Educational mission values (reference to the terms you just created before). Add different set of terms to each so that there is one unit with two terms, one with just one, one without any etc. This is to make sure that the High school search paragraph works with all cases.
9. Publish the units. Also make sure that their translations are published.
10. Go to any landing page that you have on your local or create a new one. There should now be a new paragraph that you can add called High School search. Select it and fill in the information required and pick the high school units that you edited for it.
11. Save the landing page and view it. The layout should look similar to this https://app.abstract.com/share/3fda63b5-eb44-4e2c-987b-19e0b888cc14 with the exception that there is no filter for the "Valitse yleislukio tai aikuislukio" in the layouts. This was omitted from then paragraph until the feature has been tested by the actual users and we know how it should be implemented.
12. Check that filtering the high schools results into expected behaviour. The paragraph should be very similar with the unit search paragraph with the exception that the units can be filtered with different filters and there is no pager - all the high schools should be visible on the listing always.
13. Translate the lading page to all languages and make sure all the translations are in place.

Also check this PR: https://github.com/City-of-Helsinki/drupal-hdbt/pull/213
